object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -313,15 +317,31 @@ private boolean isBuilderBuildMethod(
    *     Immutable type
    */
   private Set<String> getAllBuilderSetterMethodNames(Element builderElement) {
-    return builderElement.getEnclosedElements().stream()
+    AnnotatedTypeMirror builderType = getAnnotatedType(builderElement);
+    return getAllSupertypes((Symbol) builderElement).stream()
+        .flatMap(e -> e.getEnclosedElements().stream())
         .filter(
             e -> {
```

Comment:
This lambda is long enough that I think it ought to be its own method so that you can name and document it. As-is, this method is really hard to read.